src/parser/jobs/brd/modules/Snapshots.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @author Ririan
+ */
+import React from 'react'
+import {Table} from 'semantic-ui-react'
+import Module from 'parser/core/Module'
+import {getDataBy} from 'data'
+import STATUSES from 'data/STATUSES'
+import ACTIONS from 'data/ACTIONS'
+import {ActionLink, StatusLink} from 'components/ui/DbLink'
+
+export default class Snapshots extends Module {
+	static handle = 'snapshots'
+	static dependencies = [
+		'util',
+		//AdditionalStats module is needed because it handles adding snapshots to events.
+		'additionalStats', // eslint-disable-line @xivanalysis/no-unused-dependencies
+	]
+
+	_snapshotEvents = []
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {
+			by: 'player',
+			abilityId: [ACTIONS.CAUSTIC_BITE.id, ACTIONS.STORMBITE.id, ACTIONS.IRON_JAWS.id],
+		}, this._onSnapshot)
+
+	}
+
+	_onSnapshot(event) {
+		this._snapshotEvents.push(event)
```

Comment:
Minor, but this can be in-lined in the constructor if you want with a lambda:
```js
this.addHook('cast', {...}, event => this._snapshotEvents.push(event))
```